### PR TITLE
New longevity

### PR DESCRIPTION
**Pull request by roydahan**
Not ready to merge yet, but most of the code is there.
Leftovers after test completed:
- Rename longevity-staging to something better.
- Add more stress CMDs to Yaml.
- Add the load distribution support to stress_cmd same as did for prepare_write_cmd.
- More tuning of threads and population after the test run long enough.
- Change to 1000 keyspaces.
- Test with more than one table (10?) per keyspace.

### CODE REVIEW

File: tests/longevity-staging.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+test_duration: 10080
+prepare_write_cmd: "cassandra-stress write cl=QUORUM duration=48h -schema 'replication(factor=3)' -port jmx=6868 -mode cql3 native -rate threads=10 -pop seq=1..10000000"
+# stress_cmd: ["cassandra-stress mixed cl=QUORUM duration=10080m -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=20 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..10000000 -log interval=5",
+#             "cassandra-stress counter_write cl=QUORUM duration=10080m -schema 'replication(factor=3) compaction(strategy=DateTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1 -pop seq=1..10000000",
+#             "cassandra-stress user profile=/tmp/cs_mv_profile.yaml ops'(insert=3,read1=1,read2=1,read3=1)' cl=QUORUM duration=10080m -port jmx=6868 -mode cql3 native -rate threads=10"]
+# stress_read_cmd: ["cassandra-stress read duration=5760m -mode cql3 native -rate threads=10 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..10000000 -port jmx=6868 -log interval=5",
```
Comment (larisau) [inline tests/longevity-staging.yaml]: this test runs only prepare_write?

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1705,6 +1705,8 @@ def node_setup(node):
                 node.remoter.run("echo 'export PATH=$PATH:/usr/local/go/bin' >> $HOME/.bashrc")
                 node.remoter.run("source $HOME/.bashrc")
                 node.remoter.run("go get github.com/pdziepak/scylla-bench")
+                node.remoter.run(cmd='sudo sed -i -e "s/-server -ea/-Xms4096m -Xmx8192m -server -ea/g" /bin/cassandra-stress')
```
Comment (roydahan) [inline sdcm/cluster.py]: TEMP

File: tests/longevity-staging.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        gce_root_disk_size_db: 50
+        gce_n_local_ssd_disk_db: 8
+        gce_instance_type_loader: 'n1-standard-8'
+        gce_root_disk_type_loader: 'pd-standard'
+        gce_n_local_ssd_disk_loader: 0
+        gce_instance_type_monitor: 'n1-standard-4'
+        gce_root_disk_type_monitor: 'pd-standard'
+        gce_root_disk_size_monitor: 50
+        gce_n_local_ssd_disk_monitor: 0
+        scylla_repo: 'https://s3.amazonaws.com/downloads.scylladb.com/rpm/unstable/centos/branch-1.7/17/scylla.repo'
```
Comment (amoskong) [inline tests/longevity-staging.yaml]: Let's use SCYLLA_REPO_FILE_URL.

File: longevity_test.py [hunk 4]
```diff
@@ -41,21 +42,37 @@ def test_custom_time(self):
...
+            if pre_create_schema:
+                self._pre_create_schema()
+            if keyspace_num > 1 and self.params.get('round_robin', default='').lower() == 'true':
+                self.log.debug("Using round_robin for multiple Keyspaces...")
+                for i in xrange(1, keyspace_num):
+                    keyspace_name = 'keyspace{}'.format(i)
+                    write_queue.append(self.run_stress_thread(stress_cmd=prepare_write_cmd, keyspace_name=keyspace_name))
+                    time.sleep(5)
+            else:
+                write_queue = self.run_stress_thread(stress_cmd=prepare_write_cmd, keyspace_num=keyspace_num)
```
Comment (amoskong) [inline longevity_test.py]: `write_queue` should always be a list, you read it by for loop in line 67.

Comment (roydahan) [reply longevity_test.py]: good catch

File: tests/longevity-staging.yaml (hunk 3 above)
Comment (roydahan) [reply tests/longevity-staging.yaml]: Done (on the new file).
Keeping this file as a general staging file.

**Comment by roydahan** [comment]
Waiting for test to complete, them I'll merge

**Pull request closed without merging**